client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,168 @@
+<script lang="ts">
+  import { doc, getDoc } from "firebase/firestore";
+  import { getPrivatePlayerCollection, lobbyCollection } from "$lib/firebase/firestore-collections";
+  import { authStore as user } from "$stores/auth";
+  import type { PrivatePlayer } from "../../../functions/src/firestore-types/lobby";
+  import { onMount } from "svelte";
+
+  export let code: string;
+
+  let end: string = "";
+  let aliveCount = 0;
+  let aliveCatCount = 0;
+  let catfishDisplayname = "";
+  let currentRole: PrivatePlayer; // the player's current role
+  let lobbyRoles: PrivatePlayer; // cycle through the lobby and retrieve their roles
+
+  function lobbyReturn() {
+      // TODO: change gamestate back to lobby
```

Comment:
we probobly don't want to do that on the client side

Reply:
Does this mean I should make the function under the functions folder or under lib/firebase?

Reply:
it should be in firebase functions aka `/functions/src/` aka server side